extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+<?php
+
+/**
+ * Default message cache (based on MW Message Cache)
+ * Stores messages from messaging.wikia.com
+ */
+class DefaultMessagesCache {
+	// Holds loaded messages that are defined in MediaWiki namespace.
+	var $mCache;
+
+	var $mUseCache, $mDisable, $mExpiry;
+
+	// Variable for tracking which variables are loaded
+	var $mLoadedLanguages = array();
+
+	var $mLocalMessageCache = false;
+	var $mLocalMessageCacheSerialized = true;
+
+	function __construct( &$memCached, $useDB, $expiry ) {
+		global $wgLocalMessageCache;
+
+		$this->mUseCache = !is_null( $memCached );
+		$this->mMemc = &$memCached;
+		$this->mDisable = !$useDB;
+		$this->mExpiry = $expiry;
+
+		if( $wgLocalMessageCache !== false ) {
+			$this->mLocalMessageCache = $wgLocalMessageCache;
+		}
+	}
+
+	private function memcKey( /*... */ ) {
+		global $wgDefaultMessagesDB;
+		$args = func_get_args();
+		$key = $wgDefaultMessagesDB . ':' . implode( ':', $args );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$key". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)